lib/runner/variant-reassignment.js

```diff
@@ -152,8 +159,33 @@ export default class VariantReassignment {
     console.log(transaction)
   }
 
-  _selectExistingProductsUsingSkus () {
-    return []
+  _selectProductDraftsForReassignment (productDrafts, existingProducts) {
+    const skuToProductMap = new Map()
+    existingProducts.forEach((p) => {
+      skuToProductMap.set(p.masterVariant.sku, p)
+      p.variants.forEach(v => skuToProductMap.set(v.sku, p))
+    })
+    return productDrafts.filter(productDraft =>
+      this._isReassignmentNeeded(productDraft, skuToProductMap)
+    )
+  }
+
+  /**
+   * Product draft needs reassignment in these cases:
+   * 1. more than 1 product matches the draft's SKUs
+   * 2. or CTP product does not have exact SKU match with product draft
+   * 3. or product type is not the same
+   */
+  _isReassignmentNeeded (productDraft, skuToProductMap) {
+    const productSet = new Set()
+    const skus = [productDraft.masterVariant.sku]
+      .concat(productDraft.variants.map(v => v.sku))
+    skus.forEach(sku => productSet.add(skuToProductMap.get(sku)))
+    if (productSet.size === 1) {
```

Comment:
Maybe we should add here some check when there is no product for given productDraft SKUs? (when we sent a fresh new product to product-import)

Reply:
https://github.com/commercetools/commercetools-node-variant-reassignment/pull/12/commits/ecc17de78f6d3e174cbffe672e6e8ef6b0b0aec3